src/reanimated2/animation/springUtils.ts

```diff
@@ -151,6 +158,56 @@ export function initialCalculations(
   }
 }
 
+/** We make an assumption that we can manipulate zeta without changing duration of movement.
+ *  According to theory this change is small and tests shows that we can indeed ignore it.
+ */
+export function scaleZetaToMatchClamps(
+  animation: SpringAnimation,
+  clamp: [number, number]
+) {
+  'worklet';
+  const { zeta, toValue, startValue } = animation;
+  if (Number(toValue) === startValue) {
+    return zeta;
+  }
+
+  const [lowerBound, upperBound] =
+    Number(toValue) - startValue > 0 ? clamp : [clamp[1], clamp[0]];
```

Comment:
You can cast `toValue` to a number once and assign it to a new variable, rather than casting it every time you use it.